### ISSUE TITLE
[Bug]: Login issue where the password field does not initially accept the 8 characters

### ISSUE THREAD

**Issue by gbowne1** [gbowne1/reactsocialnetwork#229]
### Describe the Bug

This may or may not be a thing, but I am getting a minor bug where I am entering the test user's credentials.

Initially when I enter the password, it doesn't like to accept the password and I get the red border and it says Password must be 8 characters, and in this case the test users' password is totally 10 characters long.  

This happens Especially if I use the test user's saved creds from the browser. 

This also is sometimes intermittent.  It works totally if I have been to the page before.

I have to go click on Login for it to actually recognize the 10 character password as the right length.

once the user gets to the 8th ~ 9th character in their entry the field the red border and warning should dissapear as well as not immediately appear when using a saved item.

As you can see in the screen shot, there is totally 10 characters and the border/warning still appears.

### Steps to Reproduce

Start the development server by

cd to /client
run npm start

and in another terminal
cd to /server

wait for the dev server to start.

Get to the login screen and try entering the test users' creds.  you should see the screenshot below.

### Screenshots

![Screenshot 2023-05-25 at 15-24-43 React Social Media App](https://github.com/gbowne1/reactsocialnetwork/assets/47549872/c1d1fc7c-69d7-4d3c-a7bf-9d24a494240b)

### Which device are you using?

Linux VM Server

### Which operating system are you using?

Linux x64, Debian Base

### Which browser are you using?

Firefox 112

### Contribute

- [ ] I am willing to contribute and submit a pull request

**Comment by manuel12** [comment]
Will investigate this.

**Comment by manuel12** [comment]
@gbowne1 Is this currently happening? I'm having difficulty trying to reproduce the issue :/

Just to explain how the current functionality is now, after using now front-end and back-end:

- localStorage is not used anymore for storing testuser credentials, these are stored in the backend db by default when the db is created, hence why these credentials are always available.
- The validation is only run when the submit button(login/register) is pressed, so right now functionality is not that the red warning label around inputs will automatically disappear if say, you have 7 characters in password input and then you add an 8th, in this case red label will remain. Instead once you click login/register the red label will disappear, as a new validation will be ran. This is to prevent all inputs being red from the beginning, because if we run validation when page is loaded and all inputs are empty then all inputs will be red.

In any case I will keep trying to reproduce the issue and fix it if I get it, if there's anything more I can do to reproduce please write me here in a comment :)

**Comment by gbowne1** [comment]
it only happens sometimes.  I have the test creds saved in my browser.  When applied it automatically turns the field red like shown.  If I go back through, like refresh, log out, then try logging in again, this does not happen.  It's intermittent.

**Comment by manuel12** [comment]
I will add an automated test that checks this, this way anytime anyone runs the e2e tests this will be checked, adn we will catch it if happens again.

**Comment by gbowne1** [comment]
Good idea.  Just curious that a ten character autocompleted password would turn it red and display the error.

**Comment by gbowne1** [comment]
I still have not been able to nail down whats causing this either.

**Comment by gbowne1** [comment]
@manuel12 have we solved this or is this stil pretty much a non issue?

**Comment by manuel12** [comment]
> @manuel12 have we solved this or is this stil pretty much a non issue?

Need to check it again: I remember in the past I managed to reproduce this in Firefox when the browser autocompleted your password. I will look into this soon.

**Comment by gbowne1** [comment]
Ok great thanks @manuel12

**Comment by Adithya12321** [comment]
Just wanted to add to the discussion.

Even though both username and password appear filled in visually, only the password is actually reflected in the userData state, as shown in the attached console log screenshot. This seems to happen because browser autofill doesn’t reliably trigger the onChange event, especially for the username input.

One way to address this could be to add a useEffect that checks for and syncs autofilled values on mount or when the form loads. This would help ensure state and validation logic work correctly even with autofilled data.

I've attached some images for reference.

1. Click on the password input field.
2. Use the browser’s saved credentials to autofill the form.
![Image](https://github.com/user-attachments/assets/e03d7ec9-365c-42c5-9dac-2e470f96c17c)

3. Check the userData object to verify if the values have been updated correctly.
![Image](https://github.com/user-attachments/assets/3c1bb2a8-3bb9-4498-960e-f338f2e4a36e)

4. Enter values in Email section. Now we can see that the username value is updated in the userData object. Email value has not been updated still.
![Image](https://github.com/user-attachments/assets/6975f202-9b74-493c-9ce8-c6b66548720d)
